docs: update docs inline latest syntax changes (#4686)
confluentinc/ksql: docs/concepts/collections/streams.md

and doesn't create anything new. 

```sql
CREATE STREAM publications (
    author VARCHAR KEY, 
    title VARCHAR
  ) WITH (
    kafka_topic = 'publication_events',
    value_format = 'avro'
  );
```

Because the topic already exists, you do not need to specify the number of partitions.

> It's important that the columns we've defined match the data in the existing topic.

Reviewer: ```suggestion
It's important that the columns you've defined match the data in the existing topic.
```

In this case, the message would need a `KAFKA` serialized `VARCHAR` in the message key
and an `AVRO` serialized record containing a `title` field in the message value.

> If both the `author` and `title` columns where in the message value we could write: 

Reviewer: ```suggestion
If both the `author` and `title` columns are in the message value, you can write: 
```
